src/librustc_save_analysis/external_data.rs

```diff
@@ -64,6 +65,38 @@ impl SpanData {
     }
 }
 
+/// Represent an arbitrary attribute on a code element
+#[derive(Clone, Debug, RustcEncodable)]
+pub struct Attribute {
+    value: String,
+    span: SpanData,
+}
+
+impl Lower for ast::Attribute {
+    type Target = Attribute;
+
+    fn lower(mut self, tcx: TyCtxt) -> Attribute {
+        // strip #[] and #![] from the original attributes
+        self.style = ast::AttrStyle::Outer;
```

Comment:
Heh, this is a neat little hack :-)

I think the comment could explain more clearer what is happening here though